sklearn/tree/_utils.pxd

```diff
@@ -104,3 +105,49 @@ cdef class PriorityHeap:
                   double impurity, double impurity_left,
                   double impurity_right) nogil
     cdef int pop(self, PriorityHeapRecord* res) nogil
+
+# =============================================================================
+# WeightedPQueue data structure
+# =============================================================================
+
+# A record stored in the WeightedPQueue
+cdef struct WeightedPQueueRecord:
+    DOUBLE_t data
+    DOUBLE_t weight
+
+cdef class WeightedPQueue:
+    cdef SIZE_t capacity
+    cdef SIZE_t array_ptr
+    cdef WeightedPQueueRecord* array_
+
+    cdef bint is_empty(self) nogil
+    cdef SIZE_t size(self) nogil
+    cdef int push(self, DOUBLE_t data, DOUBLE_t weight) nogil
+    cdef int remove(self, DOUBLE_t value, DOUBLE_t weight) nogil
+    cdef int pop(self, DOUBLE_t* data, DOUBLE_t* weight) nogil
+    cdef int peek(self, DOUBLE_t* res, DOUBLE_t* weight) nogil
```

Comment:
good catch, I wanted to harmonize them all to `data`, but I suppose I missed one :)